src/StoreKeeper/WooCommerce/B2C/Frontend/Handlers/AddressFormHandler.php

```diff
@@ -0,0 +1,182 @@
+        $locale['NL']['postcode']['label'] = __('Postcode / ZIP & House number', I18N::DOMAIN);
+        $locale['NL']['postcode']['placeholder'] = __('Postcode / ZIP', I18N::DOMAIN);
+
+        $locale['NL']['address_1'] = [
+            'placeholder' => __('Street name', I18N::DOMAIN),
+            'priority' => 55,
+        ];
+
+        return $locale;
+    }
+
+    public function customSelectors(array $fieldSelectors): array
+    {
+        $fieldSelectors[self::HOUSE_NUMBER_FIELD] = '#billing_'.self::HOUSE_NUMBER_FIELD.'_field, #shipping_'.self::HOUSE_NUMBER_FIELD.'_field';
+
+        return $fieldSelectors;
+    }
+
+    public function customAddressFormats(array $formats): array
+    {
+        $formats['NL'] = "{company}\n{name}\n{address_house_number} {address_1}\n{address_2}\n{postcode} {city}\n{country}";
+
+        return $formats;
+    }
+
+    public function addCustomAddressArguments(array $address, int $customerId, string $addressType): array
+    {
+        if (0 !== $customerId) {
+            $customer = new \WC_Customer($customerId);
+            $houseNumber = $customer->get_meta($addressType.'_address_house_number', true);
+            if (!empty($houseNumber)) {
+                $address['address_house_number'] = $houseNumber;
+            } else {
+                $address['address_house_number'] = '';
+            }
+        }
+
+        return $address;
+    }
+
+    public function customAddressReplacements(array $replacements, array $arguments): array
+    {
+        $replacements['{address_house_number}'] = $arguments['address_house_number'] ?? '';
+
+        return $replacements;
+    }
+
+    protected function validateStreet(string $addressType, array $inputs): void
+    {
+        $countryKey = $addressType.'_country';
+
+        if (isset($inputs[$countryKey]) && AddressSearchEndpoint::DEFAULT_COUNTRY_ISO === $inputs[$countryKey]) {
+            try {
+                $postCodeKey = $addressType.'_postcode';
+                $houseNumberKey = $addressType.'_address_house_number';
+
+                if (isset($inputs[$postCodeKey])) {
+                    $postCode = $inputs[$postCodeKey];
+                }
+
+                if (isset($inputs[$houseNumberKey])) {
+                    $houseNumber = $inputs[$houseNumberKey];
+                }
+                AddressSearchEndpoint::validateAddress($postCode, $houseNumber);
+            } catch (\Throwable $throwable) {
+                wc_add_notice(sprintf(__('Invalid %s postcode or house number', I18N::DOMAIN), $addressType), 'error');
+            }
+        }
+    }
+
+    /**
+     * Validate postcode and house number if country is NL. Regex is already handled by WooCommerce validation.
+     *
+     * @see https://woocommerce.github.io/code-reference/files/woocommerce-includes-class-wc-validation.html | Line 109
+     */
+    public function validateCustomFieldsForCheckout(): void
+    {
+        $inputs = $_POST;
```

Comment:
post is not sanitized

Reply:
Added sanitization on `validateStreet` method.